packages/integrations/tailwind/src/index.ts

```diff
@@ -14,13 +16,56 @@ function getDefaultTailwindConfig(srcUrl: URL) {
 	};
 }
 
-export default function (): AstroIntegration {
+async function getUserConfig(projectRoot: URL, configPath?: string) {
+	const resolvedProjectRoot = fileURLToPath(projectRoot);
+	let userConfigPath: string | undefined;
+
+	if (configPath) {
+		const configPathWithLeadingSlash = /^\.*\//.test(configPath) ? configPath : `./${configPath}`;
+		userConfigPath = fileURLToPath(new URL(configPathWithLeadingSlash, projectRoot));
+	}
+
+	return await load('tailwind', { mustExist: false, cwd: resolvedProjectRoot, filePath: userConfigPath });
+}
+
+type TailwindOptions =
+	| {
+			config?: {
+				/**
+				 * Path to your tailwind config file
+				 * @default 'tailwind.config.js'
+				 */
+				path?: string;
+				/**
+				 * Apply Astro's default Tailwind config as a preset
+				 * This is recommended to enable Tailwind across all components and Astro files
+				 * @default true
+				 */
+				applyAstroPreset?: boolean;
+			};
+	  }
+	| undefined;
+
+export default function tailwindIntegration(options: TailwindOptions): AstroIntegration {
+	const applyAstroConfigPreset = options?.config?.applyAstroPreset ?? true;
+	const customConfigPath = options?.config?.path;
 	return {
 		name: '@astrojs/tailwind',
 		hooks: {
-			'astro:config:setup': ({ config, injectScript }) => {
+			'astro:config:setup': async ({ config, injectScript }) => {
 				// Inject the Tailwind postcss plugin
-				config.styleOptions.postcss.plugins.push(tailwindPlugin(getDefaultTailwindConfig(config.src)));
+				const userConfig = await getUserConfig(config.projectRoot, customConfigPath);
+
+				if (customConfigPath && !userConfig?.value) {
+					throw new Error(`Could not find a Tailwind config at ${JSON.stringify(customConfigPath)}. Does the file exist?`);
+				}
+
+				const tailwindConfig: TailwindConfig = (userConfig?.value as TailwindConfig) ?? getDefaultTailwindConfig(config.src);
+				if (applyAstroConfigPreset) {
+					tailwindConfig.presets = [...(tailwindConfig.presets || []), getDefaultTailwindConfig(config.src)];
```

Comment:
So if I'm reading this correctly, `tailwindConfig` becomes:

```
{
  ...tailwindConfig,
  presets: [...tailwindConfig.presets, getDefaultTailwindConfig(config.src)]
}
```

does this make sense? It feels odd to put the entire default tailwind config here as a preset. If this is expected, a comment would help future readers not get confused.

Reply:
Yep, just realized this was confusing! Added a clarifying comment here, and only applied the preset when a user config is present (otherwise we're applying the default config twice).

Reply:
Update: discovered that adding a `preset` removed Tailwind's default config. Thought they would merge 😕 Used the `resolveConfig` function to merge those defaults back in! https://github.com/withastro/astro/pull/2831/commits/19f29cb5aa6e631125824d9ca65c681533a094b3